app/controllers/incomeSources/add/BusinessReportingMethodController.scala

```diff
@@ -0,0 +1,198 @@
+    } recover {
+      case ex: Exception =>
+        Logger("application").error(s"${if (isAgent) "[Agent]"}" +
+          s"Error getting BusinessReportingMethodController page: ${ex.getMessage}")
+        errorHandler.showInternalServerError()
+    }
+  }
+
+  private def handleSubmitRequest(isAgent: Boolean, id: String)(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
+    val incomeSourcesEnabled: Boolean = isEnabled(IncomeSources)
+    val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
+    val redirectUrl: Call = if (isAgent) routes.BusinessAddedController.showAgent(id) else routes.BusinessAddedController.show(id)
+    val submitUrl: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessReportingMethodController.submitAgent(id) else
+      controllers.incomeSources.add.routes.BusinessReportingMethodController.submit(id)
+
+    if (incomeSourcesEnabled) {
+      AddBusinessReportingMethodForm.form.bindFromRequest().fold(
+        hasErrors => {
+          val updatedForm = AddBusinessReportingMethodForm.updateErrorMessagesWithValues(hasErrors)
+
+          if (updatedForm.hasErrors) {
+            businessReportingMethodService.getBusinessReportingMethodDetails(id).map {
+              case Some(viewModel) =>
+                BadRequest(view(
+                  addBusinessReportingMethodForm = updatedForm,
+                  businessReportingViewModel = viewModel,
+                  postAction = submitUrl,
+                  isAgent = isAgent))
+              case None =>
+                Redirect(redirectUrl)
+            }.recover {
+              case err: InternalServerException =>
+                Logger("application").error(s"${if (isAgent) "[Agent]"}" + s"Error getting BusinessReportingMethodDetails : $err")
+                errorHandler.showInternalServerError()
+            }
+
+
+          } else {
+            businessReportingMethodService.updateIncomeSourceTaxYearSpecific(user.nino, id, updatedForm.data).map {
+              case Some(res: UpdateIncomeSourceResponseModel) =>
+                Logger("application").info(s"${if (isAgent) "[Agent]"}" + s" Updated tax year specific reporting method : $res")
+                Redirect(redirectUrl)
+              case Some(err: UpdateIncomeSourceResponseError) =>
+                Logger("application").error(s"${if (isAgent) "[Agent]"}" + s" Failed to Updated tax year specific reporting method : $err")
+                errorHandler.showInternalServerError()
+              case None =>
+                Logger("application").info(s"${if (isAgent) "[Agent]"}" + s" Updating tax year specific reporting method not required.")
+                Redirect(redirectUrl)
+            }
+
+          }
+        },
+        valid => {
+          businessReportingMethodService.updateIncomeSourceTaxYearSpecific(user.nino, id, valid).map {
```

Comment:
duplicates code from above - but code above may not be needed as per other comment..